### PR TITLE
replace Travis CI tests with GitHub Actions tests

### PR DESCRIPTION
**Pull request by CFlaniganMide**
Trying out github actions, this may drastically speed up unit tests and simplify automation.

### CODE REVIEW

**Review by CrepeGoat — APPROVED** [review]
looks good, though I'm not intimately familiar with Actions atm.

(I do really like the much more seamless inclusion of windows & OSX)

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Unit test on push and PR
+
+
+on: [push, pull_request]
```
Comment (CrepeGoat) [inline .github/workflows/unit-tests.yml:4]: On second thought, @CFlaniganMide what do you think about adding `workflow_dispatch` as an event trigger? e.g., for #82 I kind of want to be able to run tests on the initial "added tests" commit, so it'd be nice to be able to trigger this manually (see [the docs](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow) for more info)

Comment (CFlaniganMide) [reply .github/workflows/unit-tests.yml:4]: Yeah, I don't see any reason not to add manual dispatches.

Comment (CrepeGoat) [reply .github/workflows/unit-tests.yml:4]: cool. do you want to add it or should I?

Comment (CFlaniganMide) [reply .github/workflows/unit-tests.yml:4]: I'll add it.

**Review by StokesMIDE — APPROVED** [review]

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (CrepeGoat) [reply .github/workflows/unit-tests.yml]: added in 65eada6d40214121d7762dc2aa6dc627a5ae7757

**Pull request merged**